cmd/gosky/admin.go

```diff
@@ -361,33 +361,41 @@ var listReportsCmd = &cli.Command{
 		adminKey := cctx.String("admin-password")
 		xrpcc.AdminToken = &adminKey
 
-		resp, err := comatproto.AdminGetModerationReports(ctx, xrpcc, "", 100, true, "")
-		if err != nil {
-			return err
-		}
-
-		tojson := func(i any) string {
-			b, err := json.MarshalIndent(i, "", "  ")
+		var cursor string
+		for {
+			resp, err := comatproto.AdminGetModerationReports(ctx, xrpcc, cursor, 100, true, "")
 			if err != nil {
-				panic(err)
+				return fmt.Errorf("getting reports: %w", err)
 			}
 
-			return string(b)
-		}
+			if len(resp.Reports) == 0 {
+				break
+			}
 
-		for _, rep := range resp.Reports {
-			b, err := json.MarshalIndent(rep, "", "  ")
-			if err != nil {
-				return err
+			if resp.Cursor != nil {
+				cursor = *resp.Cursor
 			}
-			fmt.Println(string(b))
-			for _, act := range rep.ResolvedByActionIds {
-				action, err := comatproto.AdminGetModerationAction(ctx, xrpcc, act)
+
+			tojson := func(i any) string {
+				b, err := json.MarshalIndent(i, "", "  ")
 				if err != nil {
-					return err
+					panic(err)
```

Comment:
if err != nil {
	fmt.Printf("Failed to marshal value: %v. Error: %v\n", i, err)

instead of panic to find the error?